clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -458,13 +458,15 @@ TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) throws IOExcep
     // Using the dummy stream to read remote buffer.
     ByteBuffer inBuf = dummyStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0,
         blockInfo.length, mTachyonConf);
+    // Close the stream object.
+    dummyStream.close();
+    if (inBuf == null) {
+      return null;
+    }
     // Copy data in network buffer into client buffer.
     ByteBuffer outBuf = ByteBuffer.allocate((int) inBuf.capacity());
     outBuf.put(inBuf);
```

Comment:
Are you sure you are allowed to read "inBuf" after you close (and release) the dummyStream?